core/mongoinit.go

```diff
@@ -0,0 +1,141 @@
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ * @author: Tingyu Zeng, Dell
+ * @version: 1.0.0
+ *******************************************************************************/
+
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/dghubble/sling"
+)
+
+type MongoCredentials struct {
+	AdminUser           string `json:"admin,omitempty"`
+	AdminPasswd         string `json:"adminpasswd,omitempty"`
+	MetadataUser        string `json:"metadata,omitempty"`
+	MetadataPasswd      string `json:"metadatapasswd,omitempty"`
+	CoredataUser        string `json:"coredata,omitempty"`
+	CoredataPasswd      string `json:"coredatapasswd,omitempty"`
+	RulesengineUser     string `json:"rulesengine,omitempty"`
+	RulesenginePasswd   string `json:"rulesenginepasswd,omitempty"`
+	NotificationsUser   string `json:"notifications,omitempty"`
+	NotificationsPasswd string `json:"notificationspasswd,omitempty"`
+	SchedulerUser       string `json:"scheduler,omitempty"`
+	SchedulerPasswd     string `json:"schedulerpasswd,omitempty"`
+	LoggingUser         string `json:"logging,omitempty"`
+	LoggingPasswd       string `json:"loggingpasswd,omitempty"`
+}
+
+func credentialInStore(config *tomlConfig, secretBaseURL string, credPath string, c *http.Client) (bool, error) {
+
+	t, err := getSecret(config.SecretService.TokenFolderPath + "/" + config.SecretService.VaultInitParm)
+	if err != nil {
+		lc.Error(err.Error())
+		return false, err
+	}
+
+	s := sling.New().Set(VaultToken, t.Token)
+
+	req, err := s.New().Base(secretBaseURL).Get(credPath).Request()
+	resp, err := c.Do(req)
+	if err != nil {
+		errStr := fmt.Sprintf("Failed to retrieve credentials with path as %s with error %s", credPath, err.Error())
+		return false, errors.New(errStr)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode == http.StatusNotFound {
+		lc.Info(fmt.Sprintf("No credential path found, setting up the credentials for %s", credPath))
+		return false, nil
+	}
+
+	lc.Info(fmt.Sprintf("%s - %d", credPath, resp.StatusCode))
+
+	var result map[string]interface{}
+
+	by, _ := ioutil.ReadAll(resp.Body)
+
+	json.Unmarshal(by, &result)
+
+	credentials := result["data"].(map[string]interface{})
+
+	if len(credentials) > 0 {
+		return true, nil
+	}
+	return false, nil
+}
+
+func initMongoDBCredentials(config *tomlConfig, secretBaseURL string, c *http.Client) error {
+
+	//TODO: we only covert coredata credential for this release, the rest needs to be implemented later.
+	adminpasswd, _ := createCredential()
+	//metadatapasswd, _ := createCredential()
+	coreadatapasswd, _ := createCredential()
+	//rulesenginepasswd, _ := createCredential()
+	//notificationspasswd, _ := createCredential()
+	//schedulerpasswd, _ := createCredential()
+	//loggingpasswd, _ := createCredential()
+
+	body := &MongoCredentials{
```

Comment:
I think a much better format for this will be to have the struct be:
```go
type UserPasswd struct {
    User string
    Passwd string
}
```

and then we have a map of strings to those structs a la 
```go
map[string]UserPasswd
```

where the `string` key is the name of the service.